core/dotnet2.2/QUICKSTART.md

```diff
@@ -19,12 +19,20 @@
 
 # Quick .NET Core 2.2 Action
 
-A .NET Core action is a .NET Core class library with a method called `Main` that has the exact signature as follows:
+A .NET Core action is a .NET Core class library with a method called `Main` or `MainAsync` that has the exact signature as follows:
```

Comment:
Isn't this just a .NET convention, that async methods should end in "Async"? (https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/task-asynchronous-programming-model)

I'd argue that that should be up to the OpenWhisk user. In the end, I imagine the usefulness of this convention is that people writing code and using this function will see right away that it returns `Task` or `Task<T>`. In this case, they're coding a function that the runtime will use, so they don't get any value out of following that convention.

Right now, the runtime detects whether it needs to await the method, and if so, it does. This means the quickstarts are simpler than in these changes because they don't need to worry about following different paths depending on if they're writing an async or sync Main method.

Reply:
These are just provided as examples of how you can create it -- it isn't gospel -- the user still has the option of defining the functions full name.

Reply:
Because the user has the option to define the function name any way they like, I think the quickstarts should be about getting started as quick as possible with as few decisions to make as possible. Someone new to OpenWhisk is likely to be more interested in getting up and running easily than tweaking a function. They wouldn't even have a function yet.